template/src/app/App.tsx

```diff
@@ -11,10 +11,12 @@ const App = (): JSX.Element => {
   return (
     <SafeAreaProvider>
       <NavigationContainer>
-        <Tab.Navigator initialRouteName='Home'>
-          <Tab.Screen component={Home} key={'Home'} name={'Home'} />
-          <Tab.Screen component={Details} key={'Details'} name={'Details'} />
-        </Tab.Navigator>
+        <SafeAreaView style={{flex: 1}}>
```

Comment:
So now in this diff we are still wrapping the whole tab navigator in safe area view (which might be the right thing to do) but I note that the screen components themselves *also* wrap in SafeAreaView, so it seems a little bit dupliated but I guess allows the screen components to be used outside the tab navigator if that is desired, although since the tab navigator *is* the entire app as defined here, then it is duplicated :thinking:

Reply:
Well, indeed, I tried to avoid this scenario by trying to wrap the screens separately in materialTopTabNavigator, however that did not work properly, and this is how I ended up with it. I converted the PR to a draft so I can have a closer look again, I'm not sure if I like this. 😂

Reply:
So I finally removed the SafeAreaView wrapping TopTabNavigator. Indeed, it was a bad decision from the first place. Ultimately, I used useInsents just like you suggested and it's perfect.

Thanks, Mike!  

Reply:
No problem - that is great (that it is working) but also a slightly difficult tradeoff as it may cause flickering for people since it's non-native. I think they mention animating navigators (like drawer) or something may flicker but honestly I don't know - not enough experience in the area. It's a tough area, all these notches and cutouts and curves these days!